src/Generation/enemy_type.py

```diff
@@ -0,0 +1,56 @@
+from enum import Enum
+from Generation.generator import Generator
+
+class Rarity(Enum):
+    """
+    This enum class represents the rarity levels for enemy types.
+    """
+    Common = 1,
+    Rare = 2,
+    Epic = 3,
+    Legendary = 4
+
+class EnemyType:
+    """
+    This class represents enemy types and their characteristics.
+
+    Attributes:
+        race (str): The race of the enemy type.
+        name (str): The name of the enemy type.
+        rarity (Rarity): The rarity level of the enemy type.
+    """
+    def __init__(self, race: str, name: str, rarity: Rarity):
+        self.race = race
+        self.name = name
+        self.rarity = rarity
+        #self.description: str # TODO: do we need a description?
+
+def generate(race: str) -> list[EnemyType]:
```

Comment:
We might need to be able to control rarity of enemies. Or is the idea to simply generate a pool of random enemies beforehand?

Reply:
Yes, the latter. I think it would be a good idea the generate everything at the beginning of the game, that is not a reaction to the player, so it won't take time to do that during the game. And you can generate as many as you want, even if you don't use them later on.

And you can define the number of types in the _form_ file.

Reply:
That's a good choice, we can also "recycle" the same enemies if we want to e.g. keep a theme consistent. This would also limit the amount of calls to the API.